lib/mariaex/protocol.ex

```diff
@@ -295,63 +295,29 @@ defmodule Mariaex.Protocol do
   """
   def disconnect(_, state = %{sock: {sock_mod, sock}}) do
     msg_send(text_cmd(command: com_quit(), statement: ""), state, 0)
-    case msg_recv(state) do
-      {:ok, packet(msg: ok_resp()), _state} ->
-        sock_mod.close(sock)
-      {:ok, packet(msg: _), _state} ->
-        sock_mod.close(sock)
-      {:error, _} ->
-        sock_mod.close(sock)
-    end
-    _ = sock_mod.recv_active(sock, 0, "")
+    sock_mod.close(sock)
     :ok
   end
 
   @doc """
   DBConnection callback
   """
-  def checkout(%{buffer: :active_once, sock: {sock_mod, sock}} = s) do
-    case setopts(s, [active: :false], :active_once) do
-      :ok                       -> sock_mod.recv_active(sock, 0, "") |> handle_recv_buffer(s)
-      {:disconnect, _, _} = dis -> dis
+  def checkout(%{sock: {sock_mod, sock}} = s) do
+    case sock_mod.recv(sock, 0, 0) do
```

Comment:
This was somewhat inspired by the original `receive_active` code path. I'm not positive it maintains the same semantics, though, as I was never able to reproduce anything _except_ a `{:error, :timeout}`. I considered just skipping this phase and returning `{:ok, %{s | buffer:<<>>}}`, but on the slim chance that there _is_ data sitting in the socket, it seems worth clearing out.